### ISSUE TITLE
Phpfox_File::instance()->unlink() return false, if using CDN

### ISSUE THREAD

**Issue by Bjoern3003** [PHPfox-Official/phpfox-v4-issues#2780]
the method call in remove function for cdn, can not return result, so the unlink function always return false, because there is nothing to unlink on the phpfox server.

Please have a look at
PF.Base\include\library\phpfox\file\file.class.php L:1543

AND

PF.Site/Apps/core-cdn/Model/CDN.php L:94

Both of these have to return their result.

**Comment by kerstar** [comment]
@harrison05 Provide here instructions on how to fix this issue, because many people will be interested in fixing this bug instead of waiting for the next version of phpfox - 4.7.9 (yesterday you released version 4.7.8) if users need to update the core to update the PhpFox Storage application from the Store.

**Comment by harrison05** [comment]
@Bjoern3003 We will check this issue.

@kerstar If bugs are easy to fix, we can provide a quick solution for customers on GitHub, but not all customers are developers and not all bugs are simple. So, we recommend they create a ticket, then we can hotfix on their sites.

Regards.

**Comment by kerstar** [comment]
Ok, I understand, I created the ticket.

**Comment by Bjoern3003** [comment]
I also created a ticket

**Comment by kerstar** [comment]
Hi @harrison05 

I noticed that this is an issue marked as a core and phpFox Storage application bug.

This means that the bug will be fixed in the next version of phpfox - 4.7.9 and in the phpFox Storage application, which will be compatible only with this version of phpfox.
So this is what I was afraid...

Following your suggestion, I created the ticket, it was checked and I received information that this bug will be fixed in the update to the phpFox Storage application - probably only compatible with phpFox version 4.9.0...

You really can't help me implement this hotfix on my site?

Please, give at least instructions on how to fix it...

**Comment by kerstar** [comment]
Ok, I got the information that the apply of hotfix on my site at the moment can not be used, because it has not yet been verified. 
I will write again in the ticket after some time, if you verify this hotfix - I hope that then I will be able to help fix this bug.

**Comment by kerstar** [comment]
Hi @harrison05 

Did you fix this issue in the latest phpFox Storage 4.6.4 app update? Is this a core bug?
Can you give me instructions so that I can use the hotfix on my site? From what I learned, the next version of phpFox will appear in November and it will not be version 4.7.9 but 4.8.0, so I assume that it will have further improvements (along with new bugs).
I want to have a stable version, hence my request for hotfix.

**Comment by kerstar** [comment]
@Bjoern3003 
Maybe they gave you a hotfix for this bug? Can you share a solution here?

**Comment by harrison05** [comment]
Hi @kerstar 

> Did you fix this issue in the latest phpFox Storage 4.6.4 app update? Is this a core bug?

This is an issue of both Core and phpFox Storage, so it will be fixed and release in the next version of phpFox.

Regards.

**Comment by Bjoern3003** [comment]
@kerstar, sorry I have not get a hotfix yet.

**Comment by kerstar** [comment]
Hi @harrison05 

Come on... You really can't help @Bjoern3003  and me to apply hotfix on our site?
At least give some instructions, we will fix it ourselves (anticipating what you wrote earlier: Yes, not all users will fix this bug themselves, but @Bjoern3003  and me can fix this issue on our pages. Whoever wants will fix it instead of waiting a month for the next one phpFox version... Please provide instructions for fixing this bug.

**Comment by harrison05** [comment]
Hi @kerstar @Bjoern3003 

Ok, our developer provided me the source code to fix this issue, just simple:

```php
    //PF.Base/include/library/phpfox/file/file.class.php
    public function unlink($sSrc)
    {
        try {
            $return = Phpfox::getLib('cdn')->remove($sSrc);
        } catch (Exception $e) {
            $return = false;
        }
        if (@unlink($sSrc)) {
            $return = true;
        }
        return $return;
    }

    //PF.Site/Apps/core-cdn/Model/CDN.php
    public function remove($file)
    {
        $file = str_replace("\\", '/', str_replace(PHPFOX_DIR, '', $file));
        // check and remove on all servers
        foreach ($this->_servers as $server_id => $server) {
            $this->_send([
                'action' => 'remove',
                'file_name' => $file,
                'server_id' => $server_id,
                'cdn_key' => $server['key']
            ], true);
        }
        return true;
    }
```

Regards.

**Comment by Bjoern3003** [comment]
Thank you.

**Comment by kerstar** [comment]
Thanks!

Just replace the "public function unlink($sSrc)" function in the "/PF.Base/include/library/phpfox/file/file.class.php" file, right?
I see that the second function "public function remove($file)" in the file "/PF.Site/Apps/core-cdn/Model/CDN.php" is the same as in the last release phpFox Storage app (4.6.4).

**Comment by harrison05** [comment]
> Thanks!
> 
> Just replace the "public function unlink($sSrc)" function in the "/PF.Base/include/library/phpfox/file/file.class.php" file, right?
> I see that the second function "public function remove($file)" in the file "/PF.Site/Apps/core-cdn/Model/CDN.php" is the same as in the last release phpFox Storage app (4.6.4).

Yes, just replace function in **file.class.php** if you're using phpFox Storage 4.6.4.

Regards.

**Comment by david-phpFox** [comment]
Fixed in Core 4.7.9.
